django/utils/datastructures.py

```diff
@@ -131,18 +131,23 @@ def get(self, key, default=None):
             return default
         return val
 
-    def getlist(self, key, default=None):
-        """
-        Returns the list of values for the passed key. If key doesn't exist,
-        then a default value is returned.
-        """
+    def _getlist(self, key, default=None):
+        # Returns mutable list of values for the passed key
+        # Used internally to manipulate list of values
         try:
             return super(MultiValueDict, self).__getitem__(key)
         except KeyError:
             if default is None:
                 return []
             return default
 
+    def getlist(self, key, default=None):
+        """
+        Returns the list of values for the passed key. If key doesn't exist,
+        then a default value is returned.
+        """
+        return list(self._getlist(key, default))
```

Comment:
MultiValueDict is designed so that it will return a list for a key. Default item accessor returns last value in a list, and getlist return all values there.

In case of absence of key a provided default value is returned but default value is never forced to be anything, it's up to API user to decide.